interfaces/cython/cantera/ctml2yaml.py

```diff
@@ -0,0 +1,2188 @@
+
+        if species_node.findtext("note") is not None:
+            self.attribs["note"] = species_node.findtext("note")
+
+        thermo = species_node.find("thermo")
+        if thermo is not None:
+            thermo_model = thermo.get("model", "")
+            # This node is not used anywhere, but we don't want it to be processed by
+            # the SpeciesThermo constructor or the hkft method
+            pseudo_species = thermo.find("pseudoSpecies")
+            if pseudo_species is not None:
+                thermo.remove(pseudo_species)
+            # The IonFromNeutral species thermo node does not correspond to a
+            # SpeciesThermo type and the IonFromNeutral model doesn't have a thermo
+            # node in the YAML format. Instead, the data from the XML thermo node are
+            # moved to the equation-of-state node in YAML
+            if thermo_model.lower() == "ionfromneutral":
+                neutral_spec_mult_node = thermo.find("neutralSpeciesMultipliers")
+                if neutral_spec_mult_node is None:
+                    raise MissingXMLNode(
+                        "IonFromNeutral requires a neutralSpeciesMultipliers node",
+                        thermo,
+                    )
+                species_multipliers = FlowMap({})
+                neutral_spec_mult = clean_node_text(neutral_spec_mult_node).split()
+                for spec_mult in neutral_spec_mult:
+                    species, multiplier = spec_mult.rsplit(":", 1)
+                    species_multipliers[species] = float(multiplier)
+                if species_multipliers:
+                    self.attribs["equation-of-state"] = {
+                        "model": "ions-from-neutral-molecule",
+                        "multipliers": species_multipliers,
+                    }
+                if thermo.find("specialSpecies") is not None:
+                    self.attribs["equation-of-state"]["special-species"] = True
+            elif thermo_model.lower() == "hkft":
+                self.attribs["equation-of-state"] = self.hkft(species_node)
+            else:
+                if len(thermo) > 0:
+                    self.attribs["thermo"] = SpeciesThermo(thermo)
+
+        transport = species_node.find("transport")
+        if transport is not None:
+            self.attribs["transport"] = SpeciesTransport(transport)
+
+        self.process_standard_state_node(species_node)
+
+        electrolyte = species_node.findtext("electrolyteSpeciesType")
+        if electrolyte is not None:
+            electrolyte = self.electrolyte_species_type_mapping[electrolyte.strip()]
+            self.attribs["electrolyte-species-type"] = electrolyte
+
+        weak_acid_charge = species_node.find("stoichIsMods")
+        if weak_acid_charge is not None:
+            self.attribs["weak-acid-charge"] = get_float_or_units(weak_acid_charge)
+
+    def hkft(self, species_node: etree.Element) -> Dict[str, "HKFT_THERMO_TYPE"]:
+        """Process a species
```

Comment:
```suggestion
                raise MissingXMLNode("Coverage requires 'a'", cov_node)
```
You might want to add quotes more broadly in the error messages, but this one had me hanging for a second -- Coverage requires a *what*?